crates/ide-completion/src/context/analysis.rs

```diff
@@ -763,6 +763,13 @@ impl<'a> CompletionContext<'a> {
                 .map_or(false, |it| it.semicolon_token().is_none());
             let impl_ = fetch_immediate_impl(sema, original_file, expr.syntax());
 
+            let in_match_guard = it.parent().and_then(ast::MatchArm::cast).is_some()
+                && it
+                    .siblings_with_tokens(Direction::Prev)
+                    .filter(|it| it.kind() == T![=>])
+                    .next()
+                    .is_none();
+
```

Comment:
This works, but feels a bit wrong to me. Instead of discarding the cast result we can use it to query the `=>` token via `.fat_arrow_token()` and then look at the text ranges instead. That feels more correct to me as we don't "arbitrarily" sift through the tokens ourselves.

<details>
  <summary>Something like the following (spoilered in case you wanna tackle what I mean yourself instead)</summary>
```rs
match it.parent().and_tehn(ast::MatchArm::cast) {
  Some(arm) => arm.fat_arrow_token().map_or(true, |it| name_ref.text_range().start() < it.text_range().start())
  None = false,
}
```
  
</details>

Reply:
Okay, that makes sense. I'm definitely still developing my sense of what "feels right", thanks for the suggestions!

Reply:
It takes time to get to know the APIs in r-a and what is the right tool for the right job that's for sure :)